src/Adapter/MSTest.TestAdapter/ObjectModel/TestMethod.cs

```diff
@@ -28,6 +29,8 @@ public sealed class TestMethod : ITestMethod
     private string? _declaringClassFullName;
     private string? _declaringAssemblyName;
 
+    private static readonly ConcurrentDictionary<string, object?[]> DataDictionary = new();
```

Comment:
Notes:

- This is bad. The dictionary in theory can grow up indefinitely.
- Currently, UniqueName isn't really unique and we need a better mechanism here. One possibility is to attach a test property resulting from `Interlocked.Increment` and use that as the dictionary key. Then, instead of deserializing we look up that key.
- This PR is mostly to open a discussion about the possibilities to improve the situation for some parameterized tests broken scenarios.

@Evangelink thoughts? I feel like some of the parameterized test issues could be fixable without the need for breaking changes?

Reply:
Resolved as I switched to the Interlocked.Increment. Will add another comment with the current concerns.